core/generated-sources/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/ColumnsSelectionDsl.kt

```diff
@@ -0,0 +1,515 @@
+        } shouldBe df.select { name.firstName }
+    }
+
+    @Test
+    fun single() {
+        val singleDf = df.select { take(1) }
+        singleDf.select { all().single() } shouldBe singleDf.select { single() }
+        singleDf.select { all().single() } shouldBe singleDf.select { name }
+        singleDf.select { single() } shouldBe singleDf.select { name }
+        df.select { single { it.name().startsWith("a") } } shouldBe df.select { age }
+
+        df.select {
+            name.single {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select {
+            name.colsOf<String>().single {
+                it.any { it == "Alice" }
+            }
+        }
+
+        df.select {
+            "name".single {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+
+        df.select {
+            Person::name.single {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+
+        df.select {
+            pathOf("name").single {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+    }
+
+    @Test
+    fun col() {
+        df.select { col("age") } shouldBe df.select { age }
+        df.select { col<Int>("age") } shouldBe df.select { age }
+        df.select { col(pathOf("age")) } shouldBe df.select { age }
+        df.select { col<Int>(pathOf("age")) } shouldBe df.select { age }
+        df.select { col(Person::age) } shouldBe df.select { age }
+
+        df.select { colGroup("name").col("firstName") } shouldBe df.select { name.firstName }
+        df.select { colGroup("name").col<String>("firstName") } shouldBe df.select { name.firstName }
+        df.select { colGroup("name").col(pathOf("firstName")) } shouldBe df.select { name.firstName }
+        df.select { colGroup("name").col<String>(pathOf("firstName")) } shouldBe df.select { name.firstName }
+        df.select { colGroup("name").col(Name::firstName) } shouldBe df.select { name.firstName }
+    }
+
+    @DataSchema
+    interface FirstNames {
+        val firstName: String
+        val secondName: String?
+        val thirdName: String?
+    }
+
+    @DataSchema
+    interface MyName : Name {
+        val firstNames: FirstNames
+    }
+
+    @Test
+    fun colGroup() {
+        val firstNames by columnGroup<FirstNames>()
+        val dfGroup = df.convert { name.firstName }.to {
+            val firstName by it
+            val secondName by it.map<_, String?> { null }.asValueColumn()
+            val thirdName by it.map<_, String?> { null }.asValueColumn()
+
+            dataFrameOf(firstName, secondName, thirdName)
+                .cast<FirstNames>(verify = true)
+                .asColumnGroup(firstNames)
+        }
+
+        dfGroup.select { colGroup("name") } shouldBe dfGroup.select { name }
+        dfGroup.select { colGroup<String>("name") } shouldBe dfGroup.select { name }
```

Comment:
what is colGroup of String?

Reply:
creates a column accessor for a column group named "name"

Reply:
I mean type parameter

Reply:
`ColumnAccessor<DataRow<String>>`

Reply:
doesn't make much sense now I think of it haha

Reply:
`colGroup<Name>("name")` is better :)

Reply:
Ok, but type parameter in ColumnGroup and DataRow is a data schema, i.e. "MyDataSchema". So i think this example is misleading - String is not a DataSchema

Reply:
I agree, Name is